sdk/opendp/whitenoise/synthesizers/pytorch/nn/__init__.py

```diff
@@ -67,7 +74,7 @@ def __init__(self,
         self.pd_cols = None
         self.pd_index = None
 
-    def train(self, data):
+    def train(self, data, categorical_columns=tuple(), ordinal_columns=tuple()):
```

Comment:
We shouldn't need categorical_columns and ordinal_columns as arguments here. They're only used to indicate how preprocessing gets handled but the GAN classes should simply take a matrix that has already been transformed as input to train()

Reply:
Hi Anuj, the reason that we need categoircal_columns and oridinal_columns here for CTGAN is because ctgan needs information from transformer for sampling and conditional generator, "self.transformer = DataTransformer()
        self.transformer.fit(train_data, categorical_columns)
        train_data = self.transformer.transform(train_data)

        data_sampler = Sampler(train_data, self.transformer.output_info)" 

so transforrmed data within train step instead of transform first and then pass prepocessed data in. Or we can pass preprocessor in instead of pass categorical column in

> We shouldn't need categorical_columns and ordinal_columns as arguments here. They're only used to indicate how preprocessing gets handled but the GAN classes should simply take a matrix that has already been transformed as input to train()